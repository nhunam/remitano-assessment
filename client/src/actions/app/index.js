import { ActionTypes } from "./type";
import { createActions } from "redux-actions";
//import proxy from 'http-proxy-middleware';

export const { clearMessage } = createActions({
  [ActionTypes.CLEAR_MESSAGE]: param => ({ ...param })
});
/*
export default function(app) {
  app.use(proxy('/users/**', { target: 'https://obscure-beach-46869.herokuapp.com/' }));
  app.use(proxy('/videos/**', { target: 'https://obscure-beach-46869.herokuapp.com/' }));
  app.use(proxy('/token/**', { target: 'https://obscure-beach-46869.herokuapp.com/' }));
};
*/