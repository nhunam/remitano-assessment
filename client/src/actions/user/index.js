import { ActionTypes } from "./type";
import { createActions } from "redux-actions";

export const {
  userLogin: login,
  userLogout: logout,
  userCreate: create
} = createActions({
  [ActionTypes.USER_LOGIN]: param => ({ ...param }),
  [ActionTypes.USER_CREATE]: param => ({ ...param }),
  [ActionTypes.USER_LOGOUT]: () => ({})
});
