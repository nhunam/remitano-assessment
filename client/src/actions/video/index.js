import { ActionTypes } from "./type";
import { createActions } from "redux-actions";

export const { videoFetch: fetch, videoSharing: shareVideo } = createActions({
  [ActionTypes.VIDEO_FETCH]: param => ({ ...param }),
  [ActionTypes.VIDEO_SHARING]: param => ({ ...param })
});
