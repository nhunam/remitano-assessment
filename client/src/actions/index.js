import { create, login, logout } from "./user";
import { fetch, shareVideo } from "./video";

import { clearMessage } from "./app";

export { login, logout, create, fetch, shareVideo, clearMessage };
