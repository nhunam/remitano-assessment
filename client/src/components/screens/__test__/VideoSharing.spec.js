import React from "react";
import VideoSharingScreen from "components/screens/VideoSharing";
import { bindActionCreators } from "redux";
import { shallow } from "enzyme";
import { shareVideo } from "actions";

describe("Video Screen", () => {
  const mockDispatching = jest.fn();
  const shareVideoMock = bindActionCreators(shareVideo, mockDispatching);
  const userMock = {
    isAuthenticated: false,
    info: { email: "test", token: "test" }
  };
  const wrapper = shallow(
    <VideoSharingScreen shareVideo={shareVideoMock} user={userMock} />
  );

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should dispatch an share action when handle submit", () => {
    const event = { preventDefault: () => {} };
    wrapper.instance().handleSubmit(event);
    expect(mockDispatching).toHaveBeenCalledWith({
      payload: {
        userToken: "test",
        videoUrl: ""
      },
      type: "VIDEO_SHARING"
    });
  });

  it("should change state of videoUrl when handle change", () => {
    const event = {
      preventDefault: () => {},
      target: {
        name: "videoUrl",
        value: "value"
      }
    };
    wrapper.instance().handleChange(event);
    expect(wrapper.state().videoUrl).toEqual("value");
  });
});
