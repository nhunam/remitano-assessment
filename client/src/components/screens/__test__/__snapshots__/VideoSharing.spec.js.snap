// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Screen should render properly 1`] = `ShallowWrapper {}`;
